inspire-tree.d.ts

```diff
@@ -1,280 +1,278 @@
+    unmute(events: Array<string>): InspireTree;
 
-    class TreeNodes extends Array<TreeNode> {
-        addNode(node: any): TreeNode;
-        available(): TreeNodes;
-        blur(): TreeNodes;
-        blurDeep(): TreeNodes;
-        checked(): TreeNodes;
-        clean(): TreeNodes;
-        clone(): TreeNodes;
-        collapse(): TreeNodes;
-        collapsed(full?: boolean): TreeNodes;
-        collapseDeep(): TreeNodes;
-        constructor(tree: InspireTree);
-        constructor(tree: InspireTree, array: Array<any>|TreeNodes);
-        context(): TreeNode;
-        copy(hierarchy?: boolean): Copy;
-        deepest(): TreeNodes;
-        deselect(): TreeNodes;
-        deselectDeep(): TreeNodes;
-        each(iteratee: NodeIteratee): TreeNodes;
-        editable(full?: boolean): TreeNodes;
-        editing(full?: boolean): TreeNodes;
-        expand(): TreeNodes;
-        expanded(full?: boolean): TreeNodes;
-        expandDeep(): TreeNodes;
-        expandParents(): TreeNodes;
-        extract(predicate: string|NodeIteratee): TreeNodes;
-        filterBy(predicate: string|NodeIteratee): TreeNodes;
-        flatten(predicate: string|NodeIteratee): TreeNodes;
-        focused(full?: boolean): TreeNodes;
-        get(index: number): TreeNode;
-        hidden(full?: boolean): TreeNodes;
-        hide(): TreeNodes;
-        hideDeep(): TreeNodes;
-        indeterminate(full?: boolean): TreeNodes;
-        insertAt(index: number, object: any): TreeNode;
-        invoke(methods: string|Array<string>): TreeNodes;
-        invokeDeep(methods: string|Array<string>): TreeNodes;
-        loading(full?: boolean): TreeNodes;
-        matched(full?: boolean): TreeNodes;
-        move(index: number, newIndex: number, target: TreeNodes): TreeNode;
-        node(id: string|number): TreeNode;
-        nodes(ids?: Array<string>|Array<number>): TreeNodes;
-        pagination(): Pagination;
-        recurseDown(iteratee: NodeIteratee): TreeNodes;
-        removed(full?: boolean): TreeNodes;
-        restore(): TreeNodes;
-        restoreDeep(): TreeNodes;
-        select(): TreeNodes;
-        selectable(full?: boolean): TreeNodes;
-        selectBetween(start: TreeNode, end: TreeNode): InspireTree;
-        selectDeep(): TreeNodes;
-        selected(full?: boolean): TreeNodes;
-        show(): TreeNodes;
-        showDeep(): TreeNodes;
-        softRemove(): TreeNodes;
-        sortBy(sorter: string|NodeIteratee): TreeNodes;
-        state(key: string, val: boolean): TreeNodes;
-        stateDeep(key: string, val: boolean): TreeNodes;
-        swap(node1: TreeNode, node2: TreeNode): TreeNodes;
-        tree(): InspireTree;
-        toArray(): Array<any>;
-        visible(full?: boolean): TreeNodes;
-    }
+    // Inherited from EventEmitter2.
+    // These are manually defined because otherwise we'd have to make
+    // the eventemitter2 module a dependency purely for the typings which
+    // simply be spam for a majority of users
```

Comment:
While you're editing this, would you mind fixing a missed word here? This line should begin `would simple be...` (added the word "would").

Reply:
Sure. Although I don't really get why you want to avoid importing?

Reply:
I see that you use `EventEmitter2` as a devDependency and bundle it in the dist file, which is the `main` entry point - but why? If we happen to use `EventEmitter2` in our app in some other place (or through some other dependency) we will have that dependency in our bundle twice, instead of relying on npm's deduplication and a module loader to only bundle it once

Reply:
Fixed the typo

Reply:
It might be the smarter way to go.

Am I correct in that in order for an inspire-tree *user* to have access to the eventemitter2 typings, I would have include eventemitter2 as a production dependency? 

If a user runs `yarn add inspire-tree --dev`, the `eventemitter2` package (and therefore its typings) would only be available if it were listed in `dependencies` and not `devDependencies`?

If so, my original reasoning was due to that. Because I compile to `dist`, eventemitter2 and other bundled libs are already included, so I didn't want add a whole dep just for the typings.

If this weren't compiled, I'd have to have those other deps available anyway.

Reply:
Yes, you are correct in that. FWIW eventemitter2 is tiny so I don't think that is a huge problem:
```
  package              size     minified  gzipped
  eventemitter2@4.1.2  28.2 kB  11.2 kB   3 kB
```
But my point was that, leaving typings completely aside, I would much prefer to have the option to import the raw ES6 `tree.js` in my application, with eventemitter2 imported as a dependency, not bundled. Because that means if my application requires eventemitter2 itself somewhere, that package is only bundled once in my bundle, instead of twice. Most libraries provide both a bundled dist/minified file in the package and the ability to just import the source JS - that is currently not possible because the dependency is missing.

Reply:
I didn't notice your subsequent replies before I made mine...

EventEmitter2 and other libs are listed as a devDeps because I bundle them (primarily for AMD users, but that file is currently what gets used by CJS users).

I dislike that, majorly, so I've debated moving them into the deps category and having CJS users import my `src` files directly. Problem is, I'm using the es6 module syntax with rollup and I don't think that's compatible with CJS users, I know node is working on es6 module support though.

Reply:
That's what I want ideally, module users should be able to import my src and bundle themselves and browser users can use the `dist` files. I've debating making that change, but the only hold up is that I don't know how the es6 imports I use will affect someone without support for it.

Reply:
My 2ct on that are that this is a pure frontend library, so we don't need to care about what Node supports. In frontend dev you either include script tags (in that case you can use the bundled version) or you have your own build pipeline, which usually includes Babel to compile down to older browsers. So a consumer of the non-bundled version would be responsible for transpiling the source. Speaking for us, we have Babel in our pipeline so all we would need to do is change the import to the src files (if the dependencies were declared). I would propose to keep the `main` field as is for backwards compat, but declare the dependency, so that consumers with their own build pipeline can choose to import the source files.
But that doesn't block this PR, it would be nice to get this merged first :)

Reply:
I'll probably change the `main` field with the next major release, but for the time-being, it seems to me that adding EventEmitter2 as a dep would solve only the typings aspect. If your build pipeline is still pulling from `dist/inspire-tree.js` you're going to get the bundled version anyway?

Reply:
I can simply write
```ts
import InspireTree from 'inspire-tree/src/tree'
```
to get the source

Reply:
But yeah, changing the `main` field in the next major version sounds like a good idea to me. The bundled versions are not really made to be used with a bundler, so no point in having `main` point to them.